app/src/main/java/org/mozilla/focus/MainActivity.kt

```diff
@@ -185,6 +189,32 @@ class MainActivity : LocaleAwareAppCompatActivity(), BrowserFragmentCallbacks, U
         override fun isStartupHomepageVisible() = getBrowserToolbarProvider()?.isStartupHomepageVisible() ?: false
     }
 
+    private inner class MainActivityFragmentLifecycleCallbacks : FragmentManager.FragmentLifecycleCallbacks() {
+        override fun onFragmentViewCreated(
+            fragmentManager: FragmentManager?,
+            fragment: Fragment?,
+            view: View?,
+            savedInstanceState: Bundle?
+        ) {
+            if (fragment is NavigationOverlayFragment) {
+                setNavigationOverlayContainerIsVisible(true)
+            }
+        }
+
+        override fun onFragmentViewDestroyed(fragmentManager: FragmentManager?, fragment: Fragment?) {
+            if (fragment is NavigationOverlayFragment) {
+                setNavigationOverlayContainerIsVisible(false)
+            }
```

Comment:
These seem pretty integral to the functionality of the fragment, is there any reason they can't be in its internal callbacks?

Reply:
The NavigationOverlay probably shouldn't know about its container that is defined by the activity, much like it shouldn't handle it's own fragment transactions. I dislike that this is not centralized with the code that attaches the fragment but the restoration code unfortunately makes this impossible: we need to listen to the lifecycle.

That being said, I realized this is a micro-optimization: we attach and remove the fragment so when there is no fragment attached, it should draw transparently and not effect the user experience: I'll remove it because it's decentralized and confusing.